### PR TITLE
Replace switch statement with an object literal

### PR DESCRIPTION
**Pull request by arku**
This is a super minor refactor I did on a piece of code which I came across while working on https://github.com/babel/babel/issues/10988

Feel free to close it if you don't think it adds any value.

### CODE REVIEW

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 1]
```diff
@@ -61,30 +61,20 @@ type ParsingContext =
...
-    case "symbol":
-      return "TSSymbolKeyword";
-    case "undefined":
-      return "TSUndefinedKeyword";
-    case "unknown":
-      return "TSUnknownKeyword";
-    default:
-      return undefined;
-  }
+  const keywordTypes = {
```
Comment (arku) [inline packages/babel-parser/src/plugins/typescript/index.js]: Is this variable name fine? Or should we name it `tsKeywordTypes`?

**Comment by arku** [comment]
@JLHwung Thanks for the quick review!

Interesting points. I definitely wouldn't have thought of those. Both points can be addressed easily, I think. Would you be interested in the PR if your points are addressed? 🙂

**Comment by arku** [comment]
@JLHwung I have addressed your comments. Take a look when you get a chance.

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 2]
```diff
@@ -57,34 +57,28 @@ type ParsingContext =
...
-      return "TSStringKeyword";
-    case "symbol":
-      return "TSSymbolKeyword";
-    case "undefined":
-      return "TSUndefinedKeyword";
-    case "unknown":
-      return "TSUnknownKeyword";
-    default:
-      return undefined;
+  if (!Object.prototype.hasOwnProperty.call(tsKeywordTypes, value)) {
```
Comment (arku) [inline packages/babel-parser/src/plugins/typescript/index.js]: @JLHwung We could avoid this check by creating the `tsKeywordTypes` object with `Object.create(null)` which should have no prototype -- hence no additional prototype chain checks when we try to access a key.

Let me know which solution you prefer.

**Comment by existentialism** [comment]
I think the switch statement is fine?

**Comment by kaicataldo** [comment]
Mind explaining what problem you're trying to solve with this change? The current code seems fine to me.

**Comment by arku** [comment]
I spent quite a bit of time thinking about this but couldn't come up with a reason other than improving the readability by a tad (the `switch` statement was a bit long with more than 10 `case`s). And I admit I didn't think about the memory implications of creating an object vs using a `switch` statement.

I am going to close the PR as it doesn't add much value to the current version. Thank you all, especially @JLHwung for teaching me something, even from this trivial PR.

And sorry for the noise!

**Comment by existentialism** [comment]
@arku there is no need to apologize, thanks for opening the PR!

**Pull request closed without merging**